toasty/simple_interface.py

```diff
@@ -0,0 +1,51 @@
+from toasty import builder, collection, pyramid, multi_tan, multi_wcs
+import reproject
+
+
+def toast_fits_list(fits_list, **kwargs):
+    """
+    Process a file or a list of FITS files into a tile pyramid with
+    a common tangential projection.
+
+    Parameters
+    ----------
+    fits_list : str or list of str
+        A single path or a list of paths to FITS files to be processed.
+    kwargs
+        Settings for the `toasty` tiling process. Common
+        settings include 'hdu_index' and 'blankval'.
+
+    Returns
+    -------
+    out_dir : :class:`str`
+        The relative path to the base directory where the tiled files are located
+    image_set_pattern : :class:`str`
+        The pattern to be used to access specific tiles
+    """
+
+    if not isinstance(fits_list, list):
+        fits_list = [fits_list]
+
+    out_dir = fits_list[0].split('.')[0]

```

Comment:
To support arguments that are generators or other unusal iterables, this function should do something like:

```
fits_list = list(fits_list)
```

Otherwise the argument might be iterable but not indexable.

When, the logic here needs to be a bit more robust; consider `/path/to/dir.with.period/subdir/myfile.fits`. Probably easiest to use `os.path.splitext`. I don't think there's one clearly best naming scheme to use here, but I like to append something like `_tiled` to the name of the tile data directory.

The output directory should also be overridable via keyword argument, since I think we might commonly want to output the tiles to a temporary directory.